### PR TITLE
Fix Bug: Missing Edge Case for odd_even_transposition_parallel.py 

### PR DESCRIPTION
**Pull request by Gyan313**
### Describe your change:
This enhanced code includes fixed bugs pointed out by @RaymondDashWu. The problem with the Algorithm was that it assumed the length of the array to be sorted equal to 10 (check the 1st for loop inside the function oe_process), which is why when @RaymondDashWu tried sorting an array with length 11, the returned sorted array is not as expected. So, I just passed an argument named arr_len which contains the array's length value.
In addition to fixing this bug, this code includes exception handling for user input, and more detailed docstrings. It's now more robust and readable.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by Gyan313** [comment]
Hi, can any of the admins merge this pr, if possible✌️

**Comment by Gyan313** [comment]
> Hi, can any of the admins merge this pr, if possible✌️

Hey @tianyizheng02 can you please review this pr.❤️
THANKS

File: sorts/odd_even_transposition_parallel.py [hunk 1]
```diff
@@ -11,29 +11,40 @@
...
 
-def oe_process(position, value, l_send, r_send, lr_cv, rr_cv, result_pipe):
+def oe_process(
+    position,
+    value,
+    l_send,
+    r_send,
+    lr_cv,
+    rr_cv,
+    result_pipe,
```
Comment (cclauss) [inline sorts/odd_even_transposition_parallel.py:39]: Please add type hints.

**Comment by cclauss** [comment]
# This branch has conflicts that must be resolved
`sorts/odd_even_transposition_parallel.py`

**Comment by cclauss** [comment]
Closing awaiting_changes PRs to prepare for Hacktoberfest 2024

**Comment by surya111111** [comment]
Are you mad by chance??
Regards,


Surya Prakash.K

**Comment by cclauss** [comment]
Not mad at all.  These PRs were closed by automated scripts.  Please feel free to try again tomorrow under Hacktoberfest 2024.

**Pull request closed without merging**